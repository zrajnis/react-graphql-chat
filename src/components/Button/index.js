import React from 'react'
import PropTypes from 'prop-types'

const Button = props => {
  return (
    <button id={props.id}>
      {props.children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
  ]).isRequired,
  id: PropTypes.string,
}

export default Button
