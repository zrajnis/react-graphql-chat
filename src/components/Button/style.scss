.button {
  background: $blue;
  border: 0;
  color: $white;
  cursor: pointer;
  font-size: 22px;
  height: 40px;
  outline: inherit;
  padding: 0;
  width: 120px;

  &:hover {
    background: $dark-blue;
  }
}
