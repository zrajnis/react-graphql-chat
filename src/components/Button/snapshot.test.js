import React from 'react'
import renderer from 'react-test-renderer'
import { shallow} from 'enzyme'
import Button from 'components/Button'

describe('Buttno component snapshot', () => {
  it('renders the button', () => {
    const tree = renderer.create(
      <Button id="submitBtn" > Test text </Button>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
