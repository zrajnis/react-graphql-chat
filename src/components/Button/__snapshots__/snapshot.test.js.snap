// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttno component snapshot renders the button 1`] = `
<button
  id="submitBtn"
>
   Test text 
</button>
`;
